### ISSUE TITLE
[Feature Request] Operationnal Dashboard

### ISSUE THREAD

**Issue by azgaviperr** [frikky/Shuffle#556]
Will be great to get an Operationnal Dashboard.

Idea it yo get in one place status of every component of Shuffle.

* Executions list with capacity to abort all or from filter execution
*  List or number of running workers and status
* number of Workflow and nodes running etc

**Comment by OpenSecureCo** [comment]
Adding some of my thoughts as well:

- how many times a workflow has ran 
- health of triggers (schedules,webhook,etc.) 
- time it takes for each workflow to run 
- container logs (opensearch, orubus, etc.), 
- resource consumption on the host running shuffle

**Comment by frikky** [comment]
:fire:

**Comment by pandvan** [comment]
It would be very useful if such operation dashboard will be added to Shuffle.
Most of the "needs" were greatly explained by @azgaviperr and @OpenSecureCo 

I can only say that the most common task i'd like to do in Shuffle is have a look at executions history, with ability to filter them (by results, by time, by workflow, ...) in order to troubleshoot issues happened in the past (if they disappear from current view in workflow editor).

Any other platform observability feature would be more than welcome.

**Comment by frikky** [comment]
> It would be very useful if such operation dashboard will be added to Shuffle. Most of the "needs" were greatly explained by @azgaviperr and @OpenSecureCo
> 
> I can only say that the most common task i'd like to do in Shuffle is have a look at executions history, with ability to filter them (by results, by time, by workflow, ...) in order to troubleshoot issues happened in the past (if they disappear from current view in workflow editor).
> 
> Any other platform observability feature would be more than welcome.

Yep. We've built in pagination for most API's at this point (executions included), but are indeed missing key observability pieces (also including app execution logs)

**Comment by frikky** [comment]
We've been working on an external status page, but due to this being a need both on the open source and cloud side, I think this would better as an open API that polls internal services. This could further be used to monitor status from 3rd party sites.

@0x0elliot: Could you build an API for this? Basically checking the status on everything in semi-realtime

**Comment by 0x0elliot** [comment]
absolutely. let's sync up on this.

**Comment by frikky** [comment]
> absolutely. let's sync up on this.

Starting simple:
- Apps (create, build, delete)
- Workflows (create, run, delete)

For workflows, make sure the workflows vary greatly, and uses all triggers

**Comment by frikky** [comment]
Workflows are on the way

**Comment by 0x0elliot** [comment]
Done?

**Comment by 0x0elliot** [comment]
Definitely isn't. a handful more work left.

**Comment by 0x0elliot** [comment]
it's live for cloud :)

**Comment by frikky** [comment]
> it's live for cloud :)

Missing before it's "done": 
- Granular control of a schedule to run it onprem
- Test onprem for putting data in the database (without the minor issues)
- A timeline of uptime/downtime of different API's
- Plotting the timeline to a status page

**Comment by frikky** [comment]
We got all the info necessary for this + status of individual things like workflows/environments/auth etc. Should lead to something good for 2.0 :))

**Comment by frikky** [comment]
From calls, these seem to pop up over and over as ideas for data management + dashboards: 

```
apache echart
https://panel.holoviz.org/gallery/index.html
https://echarts.apache.org/en/index.html
https://streamlit.io/
https://www.nocobase.com/
```

Dashboards already work, BUT they're not good enough for general availability :(

**Comment by kaanvardar** [comment]
Hey! 
Where did dashboard go? 
I was able to access it a month ago.

Where can i track the shuffle features you've added or removed?

Thank you,
Great SOAR lve it
